File: 0001-Two-sum/0001-Two-sum.js
//sorted array

function twoSum(nums,target){
    let left=0;
    let right=nums.length-1;

    while(left<right){
        let sum=nums[left]+right[left];
        if(sum==target){
            return[left,right];
        }else if(sum<target){
            left++;
        }
        else{
            right--;
        }
    }
    return [];
}

// unsorted array

function twoSumUnsorted(nums,target){
    const map=new map();

    for(let i=0;i<nums.length;i++){
        let need=target-nums[i];

        if(map.has(need)){
            return [map.get(need),i];
        }

        map.set(nums[i],i);
    }

    return [];
}

File: 0001-Two-sum/0001-Two-sum.cpp
//this is work only for sorted array
vector<int>TwoSum(vector<int>nums, int target){
    int left=0;
    int right=nums.size()-1;
    vector<int>res;

    while(left<right){
       int sum=nums[left]+nums[right];
       if(sum==target){
          res.push_back(left);
          res.push_back(right);
          return res;
       }
       else if(sum<target){
           left++;
       }
       else{
        right++;
       }
    }
    return res;
}

// onsorted arry use map 

vector<int>TwoSum_1(vector<int>nums,int target){
    unordered_map<int,int>mp;
    for(int i=0;i<nums.size();i++){
        int need = target-nums[i];

        if(mp.count(need)){
            return {mp[need],i}
        }
        mp[nums[i]]=i;
    }
    return {};
}